File: JAVA PROGRAMMING/CHomeworkOOP/src/CHomeworkOOP/CAnimal.java
package CHomeworkOOP;

public class CAnimal {
    private String m_strName;
    private String m_strType;

    /**
     * Method: SetName
     * Sets the name of the animal with a maximum length of 50 characters.
     */
    public void SetName(String name) {
        if (name.length() > 50) {
            m_strName = name.substring(0, 50);
        } else {
            m_strName = name;
        }
    }

    /**
     * Method: GetName
     * @return m_strName
     */
    public String GetName() {
        return m_strName;
    }

    /**
     * Method: SetType
     * Sets the type of the animal with a maximum length of 50 characters.
     */
    public void SetType(String type) {
        if (type.length() > 50) {
            m_strType = type.substring(0, 50);
        } else {
            m_strType = type;
        }
    }

    /**
     * Method: GetType
     * @return m_strType
     */
    public String GetType() {
        return m_strType;
    }

    /**
     * Method: MakeNoise
     * Outputs a default noise for the animal.
     */
    public void MakeNoise() {
        System.out.println("Undefined");
    }
}

File: C PROGRAMMING 2/Homework4_KB/CAnimal.cpp
// --------------------------------------------------------------------------------
// Name: CAnimal
// Abstract: Class method definitions for the CAnimal class.
// --------------------------------------------------------------------------------

// --------------------------------------------------------------------------------
// Includes
// --------------------------------------------------------------------------------
#include "CAnimal.h"
#define _CRT_SECURE_NO_WARNINGS
#include <cstring>

// --------------------------------------------------------------------------------
// Name: SetName
// Abstract: Set the animal's name.
// --------------------------------------------------------------------------------
void CAnimal::SetName(const char* pstrNewName)
{
	if (pstrNewName != nullptr && strlen(pstrNewName) < 50)
	{
		strcpy_s(m_strName, pstrNewName);
	}
	else
	{
		m_strName[0] = '\0';
	}
}


// --------------------------------------------------------------------------------
// Name: GetName
// Abstract: Get the animal's name.
// --------------------------------------------------------------------------------
const char* CAnimal::GetName()
{
	return m_strName;
}


// --------------------------------------------------------------------------------
// Name: SetType
// Abstract: Set the animal's type.
// --------------------------------------------------------------------------------
void CAnimal::SetType(const char* pstrNewType)
{
	if (pstrNewType != nullptr && strlen(pstrNewType) < 50)
	{
		strcpy_s(m_strType, pstrNewType);
	}
	else
	{
		m_strType[0] = '\0';
	}
}


// --------------------------------------------------------------------------------
// Name: GetType
// Abstract: Get the animal's type.
// --------------------------------------------------------------------------------
const char* CAnimal::GetType()
{
	return m_strType;
}


// --------------------------------------------------------------------------------
// Name: MakeNoise
// Abstract: Base class placeholder for making noise.
// --------------------------------------------------------------------------------
void CAnimal::MakeNoise()
{
	cout << "(Generic animal noise)" << endl;
}
